core/tests/protractor_desktop/topicAndStoryViewer.js

```diff
@@ -206,6 +206,51 @@ describe('Topic and Story viewer functionality', function() {
     await users.logout();
   });
 
+  it('should dismiss sign-up section and load practice session page upon' +
+  'clicking the practice session card on the last state.', async function() {
+    await topicAndStoryViewerPage.get(
+      'math', 'topic-tasv-one', 'storyplayertasvone');
+
+    await topicAndStoryViewerPage.goToChapterIndex(0);
+    await explorationPlayerPage.submitAnswer('Continue', null);
+
+    await topicAndStoryViewerPage.waitForSignUpSection();
```

Comment:
Instead of creating and using the `waitForSignUpSection` function, why don't we just use the already-available `waitFor.visibilityOf()` function? The same goes for all other `waitFor...` functions created in this PR.

Reply:
@krishnarao22 however to do it that way, wouldn't I need to interact directly with the page (i.e. use an element selector in the suite file).

I had the second point of the ['Writing the tests' section](https://github.com/oppia/oppia/wiki/WebdriverIO#writing-the-tests) in my mind when I created these `waitForXYZ` functions, i.e. to avoid using element selectors in the suite file.

Please let me know if I'm missing something/misunderstood your point.

Reply:
Replaced the functions with `waitFor` methods, done.

Reply:
Replaced the functions with `waitFor` methods.

Done.

Reply:
Hi @nithinrdy,

I think I misread this file name when I did this review and thought it was a `util` file. 

Your original file was correct. Can you restore the old versions of `topicAndStoryViewer.js` and its corresponding `util` file (with the `waitFor...` functions you wrote)? I'm really sorry about this.

Please ping me when you're done and I'll approve it straight away. Apologies again.

Reply:
No worries! I've pushed a revert commit, PTAL.